include/oneapi/dpl/internal/async_impl/async_impl_hetero.h

```diff
@@ -70,8 +70,8 @@ __pattern_walk2_async(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _For
         _BackendTag{}, ::std::forward<_ExecutionPolicy>(__exec),
         unseq_backend::walk_n<_ExecutionPolicy, _Function>{__f}, __n, __buf1.all_view(), __buf2.all_view());
 
-    if constexpr (_IsSync::value)
-        __future.wait();
+    // Call no wait, wait or deferrable wait depending on _WaitMode
+    oneapi::dpl::__par_backend_hetero::__wait_future_result<_WaitMode>{}(__future_obj);
```

Comment:
I believe this is the wrong variable name `__future` vs `__future_obj`

Reply:
Fixed